test/unit/controllers/declaration/ExporterEoriNumberControllerSpec.scala

```diff
@@ -0,0 +1,241 @@
+
+        val result = controller.submit(Mode.Normal)(postRequest(incorrectForm))
+
+        status(result) mustBe BAD_REQUEST
+        checkViewInteractions()
+      }
+
+      "EORI is not provided but trader selected that it has an EORI" in {
+
+        withNewCaching(request.cacheModel)
+
+        val incorrectForm = Json.toJson(ExporterEoriNumber(eori = None, hasEori = YesNoAnswers.yes))
+
+        val result = controller.submit(Mode.Normal)(postRequest(incorrectForm))
+
+        status(result) mustBe BAD_REQUEST
+        checkViewInteractions()
+      }
+
+      "no choice is selected and no cached ExporterDetails exist" in {
+
+        withNewCaching(request.cacheModel)
+
+        val correctForm = Json.toJson(ExporterEoriNumber(eori = None, hasEori = ""))
+
+        val result = controller.submit(Mode.Normal)(postRequest(correctForm))
+
+        status(result) mustBe BAD_REQUEST
+        checkViewInteractions()
+      }
+    }
+  }
+
+  onJourney(STANDARD, SUPPLEMENTARY, OCCASIONAL, SIMPLIFIED) { request =>
+    "should return a 303 (SEE_OTHER)" when {
+      "'No' is selected" in {
+
+        withNewCaching(request.cacheModel)
+
+        val correctForm = Json.toJson(ExporterEoriNumber(eori = None, YesNoAnswers.no))
+
+        val result = controller.submit(Mode.Normal)(postRequest(correctForm))
+
+        await(result) mustBe aRedirectToTheNextPage
+        thePageNavigatedTo mustBe controllers.declaration.routes.ExporterDetailsController.displayPage()
+        checkViewInteractions(0)
+        theCacheModelUpdated.parties.exporterDetails must be(Some(ExporterDetails(EntityDetails(None, None))))
+      }
+
+      "'Yes' is selected" in {
+
+        withNewCaching(request.cacheModel)
+
+        val eoriInput = Some(Eori("GB123456789000"))
+        val correctForm = Json.toJson(ExporterEoriNumber(eori = eoriInput, YesNoAnswers.yes))
+
+        val result = controller.submit(Mode.Normal)(postRequest(correctForm))
+
+        await(result) mustBe aRedirectToTheNextPage
+        thePageNavigatedTo mustBe controllers.declaration.routes.RepresentativeAgentController.displayPage()
+        checkViewInteractions(0)
+        theCacheModelUpdated.parties.exporterDetails must be(Some(ExporterDetails(EntityDetails(eoriInput, None))))
+      }
+    }
+  }
+
+  onJourney(CLEARANCE) { request =>
+    "should return a 303 (SEE_OTHER)" when {
+      "'No' is selected" in {
+
+        withNewCaching(request.cacheModel)
+
+        val correctForm = Json.toJson(ExporterEoriNumber(eori = None, YesNoAnswers.no))
+
+        val result = controller.submit(Mode.Normal)(postRequest(correctForm))
+
+        await(result) mustBe aRedirectToTheNextPage
+        thePageNavigatedTo mustBe controllers.declaration.routes.ExporterDetailsController.displayPage()
+        checkViewInteractions(0)
+        theCacheModelUpdated.parties.exporterDetails must be(Some(ExporterDetails(EntityDetails(None, None))))
+      }
```

Comment:
Am I right that this test case is exactly the same as for other journeys? If so, I see no need to duplicate that. Only extract the tests that are different.

Reply:
This will make the tests even more unreadable all in the name of test consistency. I will do that, but under protest.

Reply:
This has been addressed. Please review ;)